### PR TITLE
[PTQ] MinMax algo improvements

### PR DESCRIPTION
**Pull request by l-bat**
### Changes

* Move `QuantizerLayerParameters` to backend-specific `ONNXQuantizerLayerParameters`
* Generalize backend-specific methods 

### Reason for changes

*  Extend MinMax algorithm to support new backends

### Related tickets

* 92354

### CODE REVIEW

File: nncf/quantization/algorithms/min_max/onnx_backend.py [hunk 1]
```diff
@@ -139,3 +132,81 @@ def get_weight_config(config: QuantizerConfig, model: onnx.ModelProto) -> Quanti
             )
 
         return config
+
+
+class ONNXQuantizerLayerParameters:
```
Comment (andrey-churkin) [inline nncf/quantization/algorithms/min_max/onnx_backend.py:137]: Is it possible to use `@dataclass` here?

File: nncf/quantization/algorithms/min_max/onnx_backend.py [hunk 2]
```diff
@@ -101,31 +111,14 @@ def mean_minmax_statistic_collector(use_per_sample_stats: bool,
...
-        onnx_graph = ONNXGraph(model)
-        onnx_node = onnx_graph.get_node_by_name(node.node_name)
-        return onnx_graph.get_weight_channel_axis(onnx_node)
-
-    @staticmethod
-    def get_activation_quantization_axis(quantizer_config: QuantizerConfig) -> Optional[int]:
-        if quantizer_config.per_channel:
-            return 1
-        return None
+    def get_weight_tensor_port_id(node: NNCFNode) -> int:
```
Comment (kshpv) [inline nncf/quantization/algorithms/min_max/onnx_backend.py:120]: Could it be None?

File: nncf/quantization/algorithms/min_max/onnx_backend.py [hunk 3]
```diff
@@ -79,7 +80,16 @@ def target_point(target_type: TargetType,
...
     def quantizer_insertion_command(target_point: ONNXTargetPoint,
-                                    parameters: QuantizerLayerParameters) -> ONNXQuantizerInsertionCommand:
+                                    quantizer_config: QuantizerConfig,
+                                    statistics: Union[MinMaxTensorStatistic, np.ndarray],
+                                    node: NNCFNode) -> ONNXQuantizerInsertionCommand:
+        if isinstance(statistics, MinMaxTensorStatistic):
+            axis = 1 if quantizer_config.per_channel else None
+            parameters = calculate_activation_quantizer_parameters(statistics, quantizer_config, axis)
+        else:
+            axis = node.metatype.weight_definitions.weight_channel_axis if quantizer_config.per_channel else None
```
Comment (kshpv) [inline nncf/quantization/algorithms/min_max/onnx_backend.py:90]: Could be `node.metatype.weight_definitions.weight_channel_axis` value None? Or what value should it have if it is MatMul

File: nncf/quantization/algorithms/min_max/onnx_backend.py (hunk 1 above)
Comment (kshpv) [reply nncf/quantization/algorithms/min_max/onnx_backend.py:137]: I actually did the same in https://github.com/openvinotoolkit/nncf/pull/1410

**Review by kshpv — CHANGES_REQUESTED** [review]

**Review by KodiaqQ — APPROVED** [review]

File: nncf/quantization/algorithms/min_max/backend.py [hunk 4]
```diff
@@ -95,13 +95,31 @@ def target_point(target_type: TargetType, target_node_name: str, port_id: int) -
 
     @staticmethod
     @abstractmethod
-    def quantizer_insertion_command(target_point: TargetPoint,
-                                    parameters: QuantizerLayerParameters) -> TransformationCommand:
+    def activation_quantizer_insertion_command(target_point: TargetPoint,
```
Comment (andrey-churkin) [inline nncf/quantization/algorithms/min_max/backend.py:98]:
```suggestion
    def create_activation_quantizer_insertion_command(target_point: TargetPoint,
```

File: nncf/quantization/algorithms/min_max/backend.py [hunk 5]
```diff
@@ -95,13 +95,31 @@ def target_point(target_type: TargetType, target_node_name: str, port_id: int) -
...
         :param target_point: Target location for the correction.
-        :param parameters: QuantizerLayerParameters instance for the command.
+        :param quantizer_config: QuantizerConfig instance for the current layer.
+        :param statistics: MinMaxTensorStatistic to calculate activation quantization parameters.
+        :return: Backend-specific TransformationCommand for the quantizer insertion operation.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def weight_quantizer_insertion_command(target_point: TargetPoint,
```
Comment (andrey-churkin) [inline nncf/quantization/algorithms/min_max/backend.py:112]:
```suggestion
    def create_weight_quantizer_insertion_command(target_point: TargetPoint,
```

**Review by andrey-churkin — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

**Review by kshpv — APPROVED** [review]

**Pull request merged**